.github/workflows/dev_ci.yml

```diff
@@ -1,82 +1,70 @@
-name: Multi-OS Python Version Test
+name: Multi-OS Complete Test
 
-on: 
+on:
   push:
     branches:
       - dev
+      - main
   pull_request:
     branches:
       - dev
+      - main
+
+env:
+  QSPPRED_TEST_EXTRAS: true
+  QSPPRED_TEST_SPLIT_UNITS: false
 
 jobs:
-  build-and-test:
+  tests-all:
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 120
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: ['3.10']
+        os: [ ubuntu-latest ] # FIXME: add macos-latest and windows-latest (removed for now due to limited debugging options)
```

Comment:
In the end, I added this FIXME and removed MacOS from the list. There are clearly some issues with multiprocessing that will need to be debugged properly, but I do not have access to a Mac at the moment to test things properly. So I propose to just test on Linux at the moment and tackle Windows and Mac later.

Reply:
Looks good to me @martin-sicho.  Indeed, having to debug this complex functionality on the GitHub actions is rather undoable. We could have a look another time